packages/metal/src/coreNamed.js

```diff
@@ -289,3 +289,48 @@ export function isString(val) {
  */
 export function nullFunction() {
 }
+
+/**
+ * Register a custom element for a given metal component.
+ *
+ * @param {String} tagName The tag name to use for this custom element.
+ * @param {!function()} ctor Metal component constructor.
+ * @param {?Array=} observedAttrs The attributes to be "observed".
+ * @return {void} Nothing.
+ */
+export function defineWebComponent(tagName, ctor, observedAttrs) {
+  class CustomElement extends HTMLElement {
+    constructor() {
+      super();
+    }
+
+    static get observedAttributes() {
+      return observedAttrs;
+    }
+
+    connectedCallback() {
+      let shadowRoot = this.attachShadow({mode: 'open'});
+      let opts = {};
+      if (Array.isArray(observedAttrs)) {
+        for (let i = 0, l = observedAttrs.length; i < l; i++) {
+          opts[observedAttrs[i]] = this.getAttribute(observedAttrs[i]);
+        }
+      }
+      this.component = new ctor(opts, shadowRoot);
```

Comment:
This will likely give us a `A constructor name should not start with a lowercase letter` linting error...